src/main/java/ui/UI.java

```diff
@@ -150,6 +155,8 @@ private void disableUI(boolean disable) {
     }
 
     private void showMainWindow(String repoId) {
+        //Check whether there are existing stored repos. If not, it's indicative of first time users of Hubturbo.
+        boolean isAFirstTimeUser = logic.getStoredRepos().isEmpty();
```

Comment:
The comment simply repeats what the statement says. What's missing is WHY this statement has to be top of the method. That was the confusion I had and that's what I'm trying to avoid in future readers.